source/slang/slang-parser.cpp

```diff
@@ -7796,13 +7862,24 @@ namespace Slang
                 modifier->keywordName = nameAndLoc.name;
                 modifier->loc = nameAndLoc.loc;
 
+
                 // Special handling for GLSLLayoutModifier
-                if (auto glslModifier = as<GLSLLayoutModifier>(modifier))
+                if (auto glslModifier = as<GLSLParsedLayoutModifier>(modifier))
+                {
+                    parser->ReadToken(TokenType::OpAssign);
+                    glslModifier->valToken = parser->ReadToken(TokenType::IntegerLiteral);
+                }
+                //Special handling for GLSLOffsetLayoutAttribute to add to the byte offset tracker at a binding location
+                else if (auto glslOffset = as<GLSLOffsetLayoutAttribute>(modifier))
                 {
-                    if (AdvanceIf(parser, TokenType::OpAssign))
+                    if (auto binding = listBuilder.find<GLSLBindingAttribute>())
                     {
-                        glslModifier->valToken = parser->ReadToken(TokenType::IntegerLiteral);
+                        parser->ReadToken(TokenType::OpAssign);
+                        glslOffset->offset = int64_t(getIntegerLiteralValue(parser->ReadToken(TokenType::IntegerLiteral)));
+                        parser->setBindingOffset(binding->binding, glslOffset->offset);
                     }
+                    else
+                        parser->diagnose(modifier->loc, Diagnostics::missingLayoutBindingModifier);
```

Comment:
Please wrap a single line with braces.

Sometimes it may look like one line but it is not when a macro magic is involved.